### PR TITLE
Add device-mapper-specific block device properties

### LINKED ISSUE #211: Add parsing of Device Mapper Slave Devices
Add functions for parsing /sys/devices/virtual/block/dm-<N>/slaves/ to determine state of Device Mapper Slave Devices.

This is earlier implemented as part of text-collector <https://github.com/prometheus-community/node-exporter-textfile-collector-scripts/pull/6/> but it makes more sense to add native support integrated in the node-exporter itself via procfs library instead of having a separate Python script as text-collector.

### PR DESCRIPTION
**Pull request by ventifus**
Expose block device-mapper properties. This will enable node_exporter to expose the device-mapper information in a node_disk_info metric. (PR on that to follow.)
Closes #211.

Signed-off-by: W. Andrew Denton <git@flying-snail.net>

### CODE REVIEW

**Review by discordianfish — APPROVED** [review]

**Comment by SuperQ** [comment]
This needs a rebase to update the fixtures file.

File: blockdevice/stats.go [hunk 1]
```diff
@@ -178,12 +178,37 @@ type BlockQueueStats struct {
...
+	Name string
+	// RqBasedSeqIOMergeDeadline determines how long (in microseconds) a request that is a reasonable merge
+	// candidate can be queued on the request queue.
+	RqBasedSeqIOMergeDeadline uint64
+	// Suspended indicates if the device is suspended (1 is on, 0 is off).
+	Suspended uint64
+	// UseBlkMq indicates if the device is using the request-based blk-mq I/O path mode (1 is on, 0 is off).
+	UseBlkMq uint64
+	// Uuid is the DM-UUID string or empty string if DM-UUID is not set.
+	Uuid string
```
Comment (discordianfish) [inline blockdevice/stats.go:195]: Use all uppercase here: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: blockdevice/stats.go [hunk 2]
```diff
@@ -398,3 +423,44 @@ func (fs FS) SysBlockDeviceQueueStats(device string) (BlockQueueStats, error) {
 	}
 	return stat, nil
 }
+
+func (fs FS) SysBlockDeviceMapperInfo(device string) (DeviceMapperInfo, error) {
+	info := DeviceMapperInfo{}
+	// files with uint64 fields
```
Comment (SuperQ) [inline blockdevice/stats.go:429]: Nit
```suggestion
	// Files with uint64 fields.
```

File: blockdevice/stats.go [hunk 3]
```diff
@@ -398,3 +423,44 @@ func (fs FS) SysBlockDeviceQueueStats(device string) (BlockQueueStats, error) {
...
+		"suspended":                      &info.Suspended,
+		"use_blk_mq":                     &info.UseBlkMQ,
+	} {
+		val, err := util.ReadUintFromFile(fs.sys.Path(sysBlockPath, device, sysBlockDM, file))
+		if err != nil {
+			return DeviceMapperInfo{}, err
+		}
+		*p = val
+	}
+	// files with string fields
```
Comment (SuperQ) [inline blockdevice/stats.go:441]: Nit:
```suggestion
	// Files with string fields.
```

**Review by SuperQ — CHANGES_REQUESTED** [review]
Minor nits, otherwise LGTM.

**Comment by ventifus** [comment]
Alright, I've capitalized all of the "Files with" comments (there were a few existing ones too).

**Comment by discordianfish** [comment]
@SuperQ PTAL

**Comment by ddpolyakov** [comment]
Hi there. Waiting for this feature a lot in a new build. @SuperQ  could you please take a look?

**Review by SuperQ — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**